dist/tools/openocd/openocd.sh

```diff
@@ -147,40 +161,9 @@ do_flash() {
             -c 'targets' \
             -c 'reset halt' \
             ${OPENOCD_PRE_FLASH_CMDS} \
-            -c 'flash write_image erase \"${HEXFILE}\"' \
-            -c 'reset halt' \
+            -c 'flash write_image erase \"${IMAGE_FILE}\" ${IMAGE_OFFSET} ${IMAGE_TYPE}' \
```

Comment:
A last nitpick... to avoid errors, shouldn't `${IMAGE_TYPE}` be deduced from the ${IMAGE_FILE} value? because then this can allow to set `ELFFILE bin`. Though I think is not a blocker for merging as is.

Reply:
Since we default to having IMAGE_TYPE empty, we defer image type deduction to OpenOCD by default. If a user does export IMAGE_TYPE=bin and IMAGE_FILE=myfile.elf, this is obviously going to cause a bricked board (sounds worse than it is, simply re-flashing with an empty IMAGE_TYPE will unbrick it). 
In my opinion we can't protect the user from every possible configuration mistake they do, we should only make sure that the default configuration always works and that it is easy use the tool.
Can we leave any kind of image type deduction as a possible follow up and let others propose changes?
I don't know how we would do it in a good way other than using tools like `file` or `readelf`..

Reply:
Yes I agree that's maybe too much overhead to avoid this kind of errors.